columnflow/calibration/cms/jets.py

```diff
@@ -503,40 +614,57 @@ def make_jme_keys(names, jec=jec_cfg, is_data=self.dataset_inst.is_data):
 # custom jec calibrator that only runs nominal correction
 jec_nominal = jec.derive("jec_nominal", cls_dict={"uncertainty_sources": []})
 
+# explicit calibrators for standard jet collections
+jec_ak4 = jec.derive("jec_ak4", cls_dict={"jet_name": "Jet"})
+jec_ak8 = jec.derive("jec_ak8", cls_dict={"jet_name": "FatJet", "propagate_met": False})
+jec_ak4_nominal = jec_ak4.derive("jec_ak4", cls_dict={"uncertainty_sources": []})
+jec_ak8_nominal = jec_ak8.derive("jec_ak8", cls_dict={"uncertainty_sources": []})
 
-# define default functions for jec calibrator
-def get_jer_file(self, external_files: DotDict) -> str:
+
+def get_jer_config_default(self) -> DotDict:
     """
-    Function to obtain external jer files.
+    Load config relevant to the jet energy resolution (JER) smearing.
 
-    By default, this function extracts the location of the jec correction files from the current
-    config instance *config_inst*:
+    By default, this is extracted from the current *config_inst*,
+    assuming the JER configurations are stored under the 'jer'
+    aux key. Separate configurations should be specified for each
+    jet collection, using the collection name as a key. For example,
+    the configuration for the default jet collection ``Jet`` will
+    be retrieved from the following config entry:
 
     .. code-block:: python
 
-        cfg.x.external_files = DotDict.wrap({
-            "jet_jerc": "/afs/cern.ch/work/m/mrieger/public/mirrors/jsonpog-integration-9ea86c4c/POG/JME/2017_UL/jet_jerc.json.gz",
-        })
-
-    :param external_files: Dictionary containing the information about the file location
-    :return: path or url to correction file(s)
-    """ # noqa
-    return external_files.jet_jerc
+        self.config_inst.x.jer.Jet
 
+    Used in :py:meth:`~.jer.setup_func`.
 
-def get_jer_config(self) -> DotDict:
+    :return: Dictionary containing configuration for JER smearing
     """
-    Load config relevant to the JER corrections.
-
-    By default, this is extracted from the current *config_inst*:
-
-    .. code-block:: python
+    jer_cfg = self.config_inst.x.jer
+
+    # check for old-style config
+    if self.jet_name not in jer_cfg:
+        # if jet collection is `Jet`, issue deprecation warning
+        if self.jet_name == "Jet":
+            logger.warning_once(
```

Comment:
Same as above, the first argument could be `f"{id(self)_depr_jec_config}"`.

Reply:
Added as `f"{id(self)_depr_jer_config}"`.